src/transform-sdk/rust/sr-types/src/lib.rs

```diff
@@ -0,0 +1,250 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! An internal crate providing shared types for Redpanda's Data Transforms Schema Registry client.
+//!
+//! If you are looking to use schema registry within transforms you probably want crate
+//! [redpanda-transform-sdk-sr](https://crates.io/crates/redpanda-transform-sdk-sr). These types are
+//! re-exported there for usage.
+
+use std::string::FromUtf8Error;
+
+use redpanda_transform_sdk_varint as varint;
+
+/// SchemaFormat is an enum that represents the different types of schemas stored in schema registry.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum SchemaFormat {
+    Avro,
+    Protobuf,
+    Json,
+}
+
+/// SchemaId is a type for the registered ID of a schema.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct SchemaId(pub i32);
+
+/// SchemaVersion is the version of a schema under a subject within the schema registry.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct SchemaVersion(pub i32);
+
+/// Reference is a way for a one schema to reference another. The details
+/// for how referencing is done are type specific; for example, JSON objects
+/// that use the key "$ref" can refer to another schema via URL. For more details
+/// on references, see the following link:
+///
+///  https://docs.redpanda.com/current/manage/schema-reg/schema-reg-api/#reference-a-schema
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Reference {
+    name: String,
+    subject: String,
+    version: SchemaVersion,
+}
+
+impl Reference {
+    /// Create a new reference from a name, subject and version of a registered schema
+    pub fn new(
+        name: impl Into<String>,
+        subject: impl Into<String>,
+        version: SchemaVersion,
+    ) -> Self {
+        Self {
+            name: name.into(),
+            subject: subject.into(),
+            version,
+        }
+    }
+
+    /// Get the name of the reference
+    pub fn name(&self) -> &str {
+        &self.name
+    }
+
+    /// Get the subject of the referenced schema
+    pub fn subject(&self) -> &str {
+        &self.subject
+    }
+
+    /// Get the version of the referenced schema
+    pub fn version(&self) -> SchemaVersion {
+        self.version
+    }
+}
+
+/// Schema is a schema that can be registered within schema registry
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Schema {
+    schema: String,
+    format: SchemaFormat,
+    references: Vec<Reference>,
+}
+
+impl Schema {
+    /// Create a new schema with a given format and references
+    pub fn new(
+        schema: impl Into<String>,
+        format: SchemaFormat,
+        references: Vec<Reference>,
+    ) -> Self {
+        Self {
+            schema: schema.into(),
+            format,
+            references,
+        }
+    }
+
+    /// Create a new avro schema with the given refrences
```

Comment:
type here

Reply:
Really appreciate the catch and review 🙏